src/main/scala/sparkz/core/api/http/ApiDirectives.scala

```diff
@@ -18,4 +20,27 @@ trait ApiDirectives extends CorsHandler with SparkzEncoding {
       else reject(AuthorizationFailedRejection)
   }
 
+  val DEFAULT_USER: String = "user"
+
+  lazy val withBasicAuth: AuthenticationDirective[String] = authenticateBasic(realm = "secure api", basicAuthentication)
+
+  def basicAuthentication(credentials: Credentials): Option[String] = {
+    settings.apiKeyHash.isEmpty match {
+      case true => Some(DEFAULT_USER)
```

Comment:
There is a test case where the `connect` endpoint is called. I assume it's passing because it falls into the first case and still returns 200 status. Can we add two test cases where the `apiKeyHash` is set and we both provide a correct and bad auth so as to reach both branches at lines 32 and 33?

Reply:
Done